scripts/prepare_selfplay_nnue_data.py

```
@@ -0,0 +1,104 @@
+            for line_number, line in enumerate(source, 1):
+                parts = line.rstrip(b"\r\n").rsplit(b" | ", 2)
+                if len(parts) != 3:
+                    raise ValueError(f"record {line_number}: expected FEN | score | result")
+                try:
+                    validate_fen(parts[0])
+                except ValueError as exc:
+                    raise ValueError(f"record {line_number}: invalid FEN: {exc}") from exc
+                try:
+                    int(parts[1])
+                except ValueError as exc:
+                    raise ValueError(f"record {line_number}: invalid score {parts[1]!r}") from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)